core/domain/skill_services.py

```diff
@@ -230,6 +232,167 @@ def get_all_skill_summaries():
     return skill_summaries
 
 
+def get_filtered_skill_dicts(status, classroom_name, keywords, sort_by):
+    """Returns all the skill summary dicts after filtering.
+
+    Args:
+        status: string. The status of the skill.
+        classroom_name: string. The classroom_name of the topic
+            to which skill is assigned to.
+        keywords: list(string). The keywords to look for
+            in the skill description.
+        sort_by: string. A string indicating how to sort the result.
+
+    Returns:
+        list(dict). The list of skill summary dicts matching the filters.
+    """
+    skill_summary_dicts = get_all_skill_summary_dicts_with_topic_and_classroom()
+
+    filtered_skill_summary_dicts = filter_skills_by_status(
+        skill_summary_dicts, status)
+    filtered_skill_summary_dicts = filter_skills_by_classroom(
+        filtered_skill_summary_dicts, classroom_name)
+    filtered_skill_summary_dicts = filter_skills_by_keywords(
+        filtered_skill_summary_dicts, keywords)
+
+    if sort_by == (
+            constants.TOPIC_SKILL_DASHBOARD_SORT_OPTIONS['DecreasingCreatedOn']):  # pylint: disable=line-too-long
+        return sorted(
+            filtered_skill_summary_dicts,
+            key=lambda x: x['skill_model_created_on'], reverse=True)
+    elif sort_by == (
+            constants.TOPIC_SKILL_DASHBOARD_SORT_OPTIONS['IncreasingUpdatedOn']):  # pylint: disable=line-too-long
+        return sorted(
+            filtered_skill_summary_dicts,
+            key=lambda x: x['skill_model_last_updated'])
+    elif sort_by == (
+            constants.TOPIC_SKILL_DASHBOARD_SORT_OPTIONS['DecreasingUpdatedOn']):  # pylint: disable=line-too-long
+        return sorted(
+            filtered_skill_summary_dicts,
+            key=lambda x: x['skill_model_last_updated'], reverse=True)
+    else:
+        return sorted(
+            filtered_skill_summary_dicts,
+            key=lambda x: x['skill_model_created_on'])
+
+
+def get_all_skill_summary_dicts_with_topic_and_classroom():
```

Comment:
I think you need to make a domain object for this. We don't want to assemble dicts ad hoc in the backend. Work with domain objects representing clear concepts (perhaps this is an AugmentedSkillSummary and conceptually you're creating it from a SkillSummary object).

Only translate to dicts in the controller layer, as close to the URL endpoint as possible and at the last possible minute.

Reply:
Done.